webapp/src/dogma/common/components/editor/FileEditor.tsx

```diff
@@ -0,0 +1,83 @@
+import { Box, Heading, Tab, TabList, TabPanel, TabPanels, Tabs } from '@chakra-ui/react';
+import Editor, { DiffEditor, OnMount } from '@monaco-editor/react';
+import { CommitChangesForm } from 'dogma/common/components/editor/CommitChangesForm';
+import { useState, useEffect, useRef } from 'react';
+
+export type MonacoEditorProps = {
+  readOnly: boolean;
+  language: string;
+  originalContent: string;
+};
+
+const FileEditor = ({ readOnly, language, originalContent }: MonacoEditorProps) => {
+  const [tabIndex, setTabIndex] = useState(0);
+  const handleTabChange = (index: number) => {
+    setTabIndex(index);
+  };
+  const [fileContent, setFileContent] = useState('');
+  const editorRef = useRef(null);
+  const handleEditorMount: OnMount = (editor) => {
+    editorRef.current = editor;
+    setFileContent(
+      language.toLowerCase() === 'json'
+        ? JSON.stringify(JSON.parse(originalContent), null, 2)
+        : originalContent,
+    );
+  };
+  useEffect(() => {
+    if (readOnly) {
+      setTabIndex(0);
+    }
+  }, [readOnly]);
+  return (
+    <Tabs variant="enclosed-colored" size="lg" index={tabIndex} onChange={handleTabChange}>
+      <TabList>
+        <Tab>
+          <Heading size="sm">{readOnly ? 'View file' : 'Edit file'}</Heading>
+        </Tab>
+        <Tab display={readOnly ? 'none' : 'visible'}>
+          <Heading size="sm">Preview Changes</Heading>
+        </Tab>
+      </TabList>
+      <TabPanels>
+        <TabPanel>
+          <Box>
+            <Editor
+              height="100vh"
+              language={language}
+              theme="vs-dark"
+              value={fileContent}
+              options={{
+                autoIndent: 'full',
+                formatOnPaste: true,
+                formatOnType: true,
+                readOnly: readOnly,
+                automaticLayout: true,
+              }}
+              onMount={handleEditorMount}
+            />
+          </Box>
+        </TabPanel>
+        <TabPanel display={readOnly ? 'none' : 'visible'}>
+          <DiffEditor
+            height="60vh"
+            language={language}
+            theme="vs-dark"
+            original={fileContent}
+            modified={editorRef?.current?.getValue()}
+            options={{
+              autoIndent: 'full',
+              formatOnPaste: true,
+              formatOnType: true,
+              diffWordWrap: 'on',
+              useTabStops: true,
+            }}
+          />
+          <CommitChangesForm />
```

Comment:
It was difficult to find this button. 😅 I thought I could see this button in `Edit File` tab. What do you think of showing  `<CommitChangesForm />` component for both `Editor` and `DiffEditor`?
Users may optionally use `DiffEditor` to check changes.

If you want to add some workflow, it would be worth renaming `Preview Changes` to `Commit Changes` so that users easily detect that they can commit the change there.